enh(mp): Squid SNMP (#872)
centreon/centreon-documentation: en/integrations/plugin-packs/procedures/applications-squid-snmp.md

### Collected metrics & status

<!--DOCUSAURUS_CODE_TABS-->

<!--Caches-Usage-->

| Metric name                      | Description                    | Unit  |
|:---------------------------------|:-------------------------------|:------|
| cache.cpu.utilization.percentage | Cpu usage: %s %%               | %     |
| cache.memory.usage.bytes         | Memory usage: %s %s            | B     |
| cache.filedescriptors.count      | Number of file descriptors: %s | count |
| cache.objects.count              | Number of object stored: %s    | count |

<!--Protocol-Stats-->

* HTTP statistics

| Metric name                    | Description          | Unit  |
|:-------------------------------|:---------------------|:------|
| http.hits.percentage           |                      | %     |
| http.errors.count              | errors: %s           | count |
| http.traffic.in.bitspersecond  | traffic in: %s %s/s  | b/s   |
| http.traffic.out.bitspersecond | traffic out: %s %s/s | b/s   |

* ICP statistics

| Metric name                   | Description          |
|:------------------------------|:---------------------|
| icp.traffic.in.bitspersecond  | traffic in: %s %s/s  |

> | icp.traffic.out.bitspersecond | traffic out: %s %s/s |

Reviewer: Missing unit column

    your own SNMPv3 credentials combo.

| Mandatory | Name             | Description                                 |
|:----------|:-----------------|:--------------------------------------------|
|           | SNMPEXTRAOPTIONS | Configure your own SNMPv3 credentials combo |

## How to check in the CLI that the configuration is OK and what are the main options for ? 

Once the plugin installed, log into your Centreon Poller CLI using the 
*centreon-engine* user account and test the Plugin by running the following 
command:

```bash
/usr/lib/centreon/plugins//centreon_squid_snmp.pl \
    --plugin=apps::squid::snmp::plugin \
    --mode=cache-usage \
    --hostname='10.0.0.1' \
    --snmp-version='2c' \
    --snmp-community='my-snmp-community' \
    --filter-counters='' \
    --warning-cpu='80' \
    --critical-cpu='90' \
    --use-new-perfdata
```

Expected command output is shown below:

```bash
OK : Cpu usage: 30 % Memory usage: 265289728 B Number of file descriptors: 45 Number of object stored: 23 | 'cache.cpu.utilization.percentage'=30%;;;0;100 
'cache.memory.usage.bytes'=265289728B;;;0; 'cache.filedescriptors.count'=45;;;0; 'cache.objects.count'=23;;;0; 
```

> This command would trigger a WARNING alarm if the *cpu* utilizationis reported 

Reviewer: ```suggestion
This command would trigger a WARNING alarm if the *cpu* utilization is reported 
```

as over 80% (`--warning-cpu='80'`) and a CRITICAL alarm if over 90% 
(`--critical-cpu='90'`).

All available options for a given mode can be displayed by adding the 

> `--help` parameter to thecommand:

Reviewer: ```suggestion
`--help` parameter to the command:
```

```bash
/usr/lib/centreon/plugins//centreon_squid_snmp.pl  \
    --plugin=apps::squid::snmp::plugin  \
    --mode=cache-usage  \
    --help
```

All available options for a given mode can be displayed by adding the 

> `--list-mode` parameter to thecommand:

Reviewer: ```suggestion
`--list-mode` parameter to the command:
```

```bash
 /usr/lib/centreon/plugins//centreon_squid_snmp.pl  \
    --plugin=apps::squid::snmp::plugin  \
    --list-mode
```

### Troubleshooting

#### UNKNOWN: SNMP GET Request : Timeout

If you get this message, you're probably facing one of theses issues:

* Your SNMP server isn't started or misconfigured
* An external device is blocking your request (firewall, ...)

#### UNKNOWN: SNMP GET Request : Cant get a single value

This message generally means that SNMP privileges are not wide enough for

> the mode/plugin to work properly.

Reviewer: ```suggestion
the mode/Plugin to work properly.
```
